panel/widgets/chat.py

```diff
@@ -963,10 +963,11 @@ def stream(
         """
         replace = entry is None
         entry = self._build_entry(entry or value, user=user, avatar=avatar)
-        if replace:
-            self._replace_placeholder(entry)
-        else:
-            self._stream(value, entry)
+        if entry:
+            if replace:
+                self._replace_placeholder(entry)
+            else:
+                self._stream(value, entry)
```

Comment:
This s a bit confusing to me; it should never be None

Reply:
It can be None due to the line below

![image](https://github.com/holoviz/panel/assets/42288570/3711bb5f-4468-4e8a-9624-e6f55f4b861b)

Reply:
We have seen use cases where the callback should not respond anything. We do this by returning None. For example in the example I had where a user called Philipp was ignored. He would not get responded to by the callback 😄

Reply:
That is maybe not the best example. But  you could imagine setting up the ChatInterface to provide multiple assistants that will respond to each other. And the logic might be easier to implement if you always "respond" to any new message. But depending on the message and user you want to "break out" by returning None.

Reply:
Thanks for clarifying.